pkg/app/pipedv1/livestatereporter/livestatereporter_test.go

```diff
@@ -0,0 +1,294 @@
+			r, err := plugin.NewPluginRegistry(
+				context.Background(),
+				[]plugin.Plugin{
+					{
+						Name: "k8s",
+						Cli: &fakePlugin{
+							pipelineStages: []*model.PipelineStage{
+								{
+									Name: "K8S_CANARY_ROLLOUT",
+								},
+								{
+									Name: "K8S_CANARY_CLEAN",
+								},
+								{
+									Name: "K8S_PRIMARY_ROLLOUT",
+								},
+							},
+						},
+					},
+					{
+						Name: "wait",
+						Cli: &fakePlugin{
+							pipelineStages: []*model.PipelineStage{
+								{
+									Name: "WAIT",
+								},
+							},
+						},
+					},
+				},
+			)
+
+			require.NoError(t, err)
+			return r
+		}(),
+		gitClient: gitClient,
+		pipedConfig: &config.PipedSpec{
+			Repositories: []config.PipedRepository{
+				{
+					RepoID: "repo-id",
+					Remote: "https://github.com/pipe-cd/examples.git",
+					Branch: "master",
+				},
+			},
+		},
+		secretDecrypter: nil,
+		workingDir:      t.TempDir(),
+		logger:          zaptest.NewLogger(t),
+	}
+
+	pr.flushSnapshots(context.Background())
+}
+
+func Benchmark_reporter_flushSnapshots(b *testing.B) {
+	gitClient, err := git.NewClient()
+	require.NoError(b, err)
+
+	pr := &reporter{
+		snapshotFlushInterval: 1 * time.Minute,
+		appLister: &fakeAPILister{
+			apps: func() []*model.Application {
+				apps := make([]*model.Application, 0, 100)
+				for i := 0; i < 100; i++ {
+					apps = append(apps, &model.Application{
+						Id:   "app-id",
+						Name: "app-name",
+						GitPath: &model.ApplicationGitPath{
+							Repo: &model.ApplicationGitRepository{
+								Id:     "repo-id",
+								Remote: "https://github.com/pipe-cd/examples.git",
+								Branch: "master",
+							},
+							Path:           "kubernetes/canary",
+							ConfigFilename: "app.pipecd.yaml",
+						},
+					})
+				}
+
+				return apps
+			}(),
+		},
+		apiClient: &fakeAPIClient{},
+		pluginRegistry: func() plugin.PluginRegistry {
+			r, err := plugin.NewPluginRegistry(
+				context.Background(),
+				[]plugin.Plugin{
+					{
+						Name: "k8s",
+						Cli: &fakePlugin{
+							pipelineStages: []*model.PipelineStage{
+								{
+									Name: "K8S_CANARY_ROLLOUT",
+								},
+								{
+									Name: "K8S_CANARY_CLEAN",
+								},
+								{
+									Name: "K8S_PRIMARY_ROLLOUT",
+								},
+							},
+						},
+					},
+					{
+						Name: "wait",
+						Cli: &fakePlugin{
+							pipelineStages: []*model.PipelineStage{
+								{
+									Name: "WAIT",
+								},
+							},
+						},
+					},
+				},
+			)
+
+			require.NoError(b, err)
+			return r
+		}(),
+		gitClient: gitClient,
+		repoMap:   make(map[string]git.Repo, 0),
+		pipedConfig: &config.PipedSpec{
+			Repositories: []config.PipedRepository{
+				{
+					RepoID: "repo-id",
+					Remote: "https://github.com/pipe-cd/examples.git",
+					Branch: "master",
+				},
+			},
+		},
+		secretDecrypter: nil,
+		workingDir:      b.TempDir(),
+		logger:          zaptest.NewLogger(b),
+	}
+
+	pr.flushSnapshots(context.Background())
```

Comment:
There is no use of `b *testing.B`; what is this benchmark to measure?

Reply:
I wanted to measure the time to execute it in parallel for reference.
I left it because It will be helpful to check the performance. WDYT?

Reply:
Okay, I got it.
Then you should use testing.B as below.

```go
for range b.N {
	pr.flushSnapshots(context.Background())
}
```

ref; https://pkg.go.dev/testing#hdr-Benchmarks

Reply:
OK, I added it.
[3adea7e](https://github.com/pipe-cd/pipecd/pull/5457/commits/3adea7ee1f20cb31f0793cf4617c427a50117a93)